### PR TITLE
Common: Support read fw image command

### PR DESCRIPTION
**Pull request by Yi-Shum**
Summary:
- Support read FW image command(NetFn: 0x38, Cmd: 0x18). 
Request: 
Byte 1:3 – MFG ID – 00A015h, LSB first
Byte 4 – target 
0x01 – BIOS FW 
Byte 5:8 – start offset, LSB first 
Byte 9 – read length 
Response: 
Byte 1 – Completion Code 
00h - Success 
Byte 2:4 – MFG ID – 00A015h, LSB first 
Byte 5:N – read value
- Modify wrong macro name from "IPMI_OEM_EVENT_TYPE_DEASSART" to "IPMI_OEM_EVENT_TYPE_DEASSERT" in yv35-hd and wc-mb.

Test Plan:
- Build code: PASS
- Command test: PASS

Log:
root@bmc-oob:~# bic-util slot1 0xe0 0x18 0x15 0xa0 0x00 0x00 0x60 0x57 0x13 0x00 0x80
15 A0 00 FD 31 9F 26 F9 34 B3 9E FB 0B CD C7 FA
0C 75 18 EF 23 74 3D D6 17 A8 C3 70 DF 1A CB A7
3F E6 D3 24 9F 66 D6 73 7F 9A 70 3F 1A F2 69 91
4F 8B 7C 5A 96 CF 92 98 4F 8B 7C 5A E4 D3 B2 7C
96 C4 7C 5A E4 D3 CA 7A EE C3 71 2D D6 A7 A9 C3
70 BF 0C CB 67 49 CC A7 45 3E AD AC E7 7E 18 57
63 7D 9C 3A 0C F7 B1 B0 7C 96 C4 7C 5A E4 D3 22
9F 96 E5 B3 24 E6 D3 22 9F 16 F9 B4 2C 9F 25 31
9F 16 F9

### CODE REVIEW

File: common/lib/util_spi.c [hunk 1]
```diff
@@ -367,6 +367,27 @@ uint8_t fw_update(uint32_t offset, uint16_t msg_len, uint8_t *msg_buf, bool sect
 	return FWUPDATE_SUCCESS;
 }
 
+int read_fw_image(uint32_t offset, uint8_t msg_len, uint8_t *msg_buf, uint8_t flash_position)
```
Comment (GoldenBug) [inline common/lib/util_spi.c:370]: Need to check if msg_buf is NULL

**Review by GoldenBug — CHANGES_REQUESTED** [review]

File: common/lib/util_spi.c (hunk 1 above)
Comment (Yi-Shum) [reply common/lib/util_spi.c:370]: OK, will check if this parameter is NULL.

**Comment by Yi-Shum** [comment]
Hi @GoldenBug,
I has modified it base on the suggestion.
Can you help to review it again? thanks.

**Comment by Yi-Shum** [comment]
Hi @GoldenBug ,
The flash should be switched back to CPU if reading the image fails.
I had add a commit to fix it.
Can you help to review it again? thanks.

**Pull request closed without merging**